File: examples/dotnet/StructDefExample/Person.cs
using LittleHorse.Sdk.Worker;

namespace StructDefExample;

[LHStructDef("person", "A person with a name and home address.")]
public class Person
{
    [LHStructField(description: "This is the first name of the person, i.e. their given name.")]
    public string FirstName { get; set; } = string.Empty;

    [LHStructField(description: "This is the last name of the person, i.e. their family name.")]
    public string LastName { get; set; } = string.Empty;

    [LHStructField(masked: true, description: "The home address of the person.")]
    public Address HomeAddress { get; set; } = new();

    public Person()
    {
    }

    public Person(string firstName, string lastName, Address homeAddress)
    {
        FirstName = firstName;
        LastName = lastName;
        HomeAddress = homeAddress;
    }

    public override string ToString()
    {
        return $"{FirstName} {LastName}";
    }
}

File: examples/java/struct-builder/src/main/java/io/littlehorse/examples/Person.java
package io.littlehorse.examples;

import io.littlehorse.sdk.worker.LHStructDef;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@LHStructDef("person")
public class Person {
    private String name;
    private String email;
    private Address address;

    public Person() {}

    public Person(String name, String email, Address address) {
        this.name = name;
        this.email = email;
        this.address = address;
    }

    @Override
    public String toString() {
        return "%s <%s> at %s".formatted(name, email, address);
    }
}
